src/modsim2/similarity/metrics.py

```diff
@@ -0,0 +1,167 @@
+import logging
+import platform
+from abc import ABC, abstractmethod
+
+import numpy as np
+import torch
+from otdd.pytorch.distance import DatasetDistance
+from sklearn import metrics
+from torch.utils.data import TensorDataset
+
+from modsim2.similarity.embeddings import EMBEDDING_FN_DICT
+
+
+class DistanceMetric(ABC):
+    def __init__(self, seed: int = 42):
```

Comment:
We want to make sure the DMPair object controls the seeding (so when the user sets up a dataset config, the seed is used across data creation, model training, and metrics calculation - we've stuck to a single seed for simplicity), so best to avoid having a default arg here to make sure we're never accidentally using this?